python/katana/local/analytics/_ksssp.pyx

```diff
@@ -192,18 +213,61 @@ def ksssp(pg, str edge_weight_property_name, size_t start_node,
         katana.local.initialize()
 
         graph = Graph(get_rdg_dataset("ldbc_003"))
-        from katana.local.analytics import ksssp
+        from katana.local.analytics import ksssp, KssspStatistics
         weight_name = "workFrom"
         start_node = 0
         report_node = 10
         num_paths = 5
-        ksssp(graph, weight_name, start_node, report_node, num_paths)
+        table = ksssp(graph, weight_name, start_node, report_node, num_paths)
+        stats = KssspStatistics(graph, weight_name, table, report_node)
+        print(stats)
 
     """
 
     cdef string edge_weight_property_name_str = bytes(edge_weight_property_name, "utf-8")
     txn_ctx = txn_ctx or TxnContext()
-    with nogil:
-        handle_result_void(Ksssp(underlying_property_graph(pg), edge_weight_property_name_str,
-                                 start_node, report_node, num_paths, is_symmetric,
-                                 underlying_txn_context(txn_ctx), plan.underlying_))
+    return handle_result_ksssp(Ksssp(underlying_property_graph(pg), edge_weight_property_name_str,
+                                  start_node, report_node, num_paths, is_symmetric,
+                                  underlying_txn_context(txn_ctx), plan.underlying_))
+
+
+cdef _KssspStatistics handle_result_KssspStatistics(Result[_KssspStatistics] res) nogil except *:
+    if not res.has_value():
+        with gil:
+            raise_error_code(res.error())
+    return res.value()
+
+
+cdef class KssspStatistics(Statistics):
+    """
+    Compute the :ref:`statistics` of a kSSSP computation on a graph
+    """
+    cdef _KssspStatistics underlying
+
```

Comment:
This class needs properties to expose the actual statistics. At the moment Python cannot actually access any of the statistical values.

Reply:
For now KssspStatistics just stores the paths as arrays and a weight for each path, any suggestions for other stats to calculate?

Reply:
No. But those stats are not exposed to python. :-) That's the issue you need to actually wrap the accessors using properties.

Reply:
There are still no statistics exposed to python on this class. However there are definitely statistics here: https://github.com/KatanaGraph/katana/blob/4370ed81177ea2793daab468d379f8db511c6334/libgraph/include/katana/analytics/k_shortest_paths/ksssp.h#L104-L105

I guess we can just omit the statistics from the Python API for the moment, but if that's the case there is no reason to include `KssspStatistics` in the Python interface at all.

Reply:
My point is that just printing the statistics doesn't seem useful. You can do that already, but being able to programmatically access them seems pretty important.